test/libsolidity/SyntaxTestHelper.cpp

```diff
@@ -0,0 +1,153 @@
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <test/libsolidity/SyntaxTestHelper.h>
+#include <test/libsolidity/AnalysisFramework.h>
+#include <test/TestHelper.h>
+#include <boost/algorithm/string/replace.hpp>
+
+using namespace std;
+using namespace boost::unit_test;
+namespace fs = boost::filesystem;
+
+namespace dev
+{
+namespace solidity
+{
+namespace test
+{
+
+void SyntaxTester::runTest(SyntaxTest const& _test)
+{
+	vector<string> unexpectedErrors;
+	auto expectations = _test.expectations;
+	auto errorList = parseAnalyseAndReturnError(_test.source, true, true, true).second;
+
+	bool errorsMatch = true;
+
+	if (errorList.size() != expectations.size ())
+		errorsMatch = false;
+	else
+	{
+		for (size_t i = 0; i < errorList.size(); i++)
+		{
+			auto const& error = errorList[i];
+			auto const& expectation = expectations[i];
+			bool const typeMatches = error->typeName() == expectation.first;
+			bool const messageMatches = errorMessage(*error) == expectation.second;
+			if (!typeMatches || !messageMatches)
+			{
+				errorsMatch = false;
+				break;
+			}
+		}
+	}
+
+	if (!errorsMatch)
+	{
+		string msg = "Test expectation mismatch.\nExpected result:\n";
+		if (expectations.empty())
+			msg += "\tSuccess\n";
+		else
+			for (auto const &expectation: expectations)
+			{
+				msg += "\t" + expectation.first + ": " + expectation.second + "\n";
+			}
+		msg += "Obtained result:\n";
+		if (errorList.empty())
+			msg += "\tSuccess\n";
+		else
+			for (auto &error: errorList)
+			{
+				msg += "\t" + error->typeName() + ": " + errorMessage(*error) + "\n";
+			}
+		BOOST_ERROR(msg);
+	}
+}
+
+std::string SyntaxTester::errorMessage(Error const &_e)
+{
+	if (_e.comment())
+	{
+		std::string msg = *_e.comment();
+		boost::replace_all(msg, "\n", "\\n");
+		return msg;
+	}
+	else
+	{
+		return "NONE";
+	}
+}
+
+int SyntaxTester::registerTests(
+	test_suite& _suite,
+	fs::path const& _basepath,
+	fs::path const& _path
+)
+{
+
+	int numTestsAdded = 0;
+	fs::path fullpath = _basepath / _path;
+	if (fs::is_directory(fullpath))
+	{
+		test_suite* sub_suite = BOOST_TEST_SUITE(_path.filename().string());
+		for (auto const& entry: boost::iterator_range<fs::directory_iterator>(
+			fs::directory_iterator(fullpath),
+			fs::directory_iterator()
+		))
+			numTestsAdded += registerTests(*sub_suite, _basepath, _path / entry.path().filename());
+		_suite.add(sub_suite);
+	}
+	else
+	{
+		_suite.add(make_test_case(
+			[fullpath] { SyntaxTester().runTest(SyntaxTestParser().parse(fullpath.string())); },
```

Comment:
Somehow, your comment about fixtures and recreating the test got lost. I'm not 100% sure what to do best, but perhaps it would be a good idea to actually create a test fixture here, can we do that?

Reply:
Ah sorry, now I read your comment on the deleted comment ;-)

Reply:
That comment was wrong anyways - the boost documentation is somewhat misleading there. It turns out that fixtures are in fact "shared" among all tests and there is "only one instance", but still for each individual test case the constructor and destructor of the fixture is called...
So the only difference between the current code and a boost fixture is that the boost fixture would do less memory allocations - I can look into using an actual boost fixture here, though.

Reply:
Yes - it was too late when I realized, that deleting a comment together with mail notifications may end up being confusing :-).

Reply:
It's fine like this, the object is allocated on the stack anyway.

Editing comments is also fun, you follow a link and the comment says something completely different :)